pandas/core/reshape/merge.py

```diff
@@ -93,7 +95,9 @@
 def merge(
     left: DataFrame | Series,
     right: DataFrame | Series,
+    # TODO: default how=cross if condition defined?
```

Comment:
I think this suggestion makes sense. so in the signature `how=None` and is resolved to `inner` unless `condition is not None` in which case it resolves to `cross`. Docstring should be sufficient to convey this behavior.

Reply:
I take this back. See here:
https://github.com/pandas-dev/pandas/pull/42964#issuecomment-915423616